### PR TITLE
Add sepolia client

### PR DESCRIPTION
**Pull request by rccsousa**
## Why
In order to test Wordle3 on Sepolia Testnet to be able to interact with the game publicly and polish smart contract interaction.
## How
- By updating wagmi configs to add a sepolia testnet bridge and refactoring smart contract interaction by using wagmi `useWriteContract` hooks.

EDIT @DavideSilva adding aditional context on other changes:
- By introducing a `reward` method on the `ERC20` to facilitate Wordle smart contract deployment by minting fresh WDT tokens and allocating to Wordle faucet/reward pool;
- Introduces some additional smart contract tests to accomodate this change as well as fork tests;

### CODE REVIEW

File: frontend/src/App.css [hunk 1]
```diff
@@ -30,7 +30,7 @@
 
 .header {
   display: flex;
-  max-width: 60%;
+  max-width: 80%;
```
Comment (rccsousa) [inline frontend/src/App.css:33]: fixes readability (was too small)

File: frontend/src/App.jsx [hunk 2]
```diff
@@ -76,45 +59,60 @@ function App() {
             setVictory(undefined)
         }
 
-        if (address) getParams(address)
-        if (!address) logout()
+        function isGuessCorrect(array) {
+            return array.every(function (element) {
+                return element.state === 2n;
+            });
+        }
```
Comment (rccsousa) [inline frontend/src/App.jsx:66]: moved places

File: frontend/src/App.jsx [hunk 3]
```diff
@@ -76,45 +59,60 @@ function App() {
...
-                // check if the latest (the one submited) is correct
-                const latest = guesses[guesses.length - 1];
-                setVictory(isGuessCorrect(latest))
-            })
-        }, 500);
-    };
+        // disabling eslint exhaustiv deps warn for playerGuesses / checkIfWon
+        // as these aren't necessary dependencies for useEffect
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [address, isConfirmed])
```
Comment (rccsousa) [inline frontend/src/App.jsx:103]: updated to listen for confirmations (when user is interacting with wallet)

File: src/ERC20.sol [hunk 4]
```diff
@@ -4,7 +4,19 @@ pragma solidity ^0.8.28;
 import {ERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
 
 contract WDLToken is ERC20 {
+    address public owner = msg.sender;
```
Comment (DavideSilva) [inline src/ERC20.sol:7]: This works? I'm not sure this is correct as `msg.sender` only has a value when called from a transaction. You can declare the variable without initialising it

**Review by DavideSilva — CHANGES_REQUESTED** [review]
Break this down into smaller PR's please. Lots of changes here that are not related to the PR title or description. It's getting messy and it's becoming difficult to review

File: src/ERC20.sol (hunk 4 above)
Comment (rccsousa) [reply src/ERC20.sol:7]: It's working, yes, but shouldn't be there. 
I'll re-review all the submited code.

File: frontend/src/App.jsx [hunk 5]
```diff
@@ -1,30 +1,30 @@
...
-
 // import components
-import {Alphabet} from './components/Alphabet/Alphabet'
+import { Alphabet } from './components/Alphabet/Alphabet'
 import { GameSpace } from './components/GameSpace/GameSpace'
-import { ApprovalModal } from './components/ApprovalModal/ApprovalModal'
+//TODO: Remove after testing import { ApprovalModal } from './components/ApprovalModal/ApprovalModal'
 import { LowFunds } from './components/LowFunds/LowFunds'
-import { AlreadyPlayed } from './components/AlreadyPlayed/AlreadyPlayed'
+//TODO: Remove after testing import { AlreadyPlayed } from './components/AlreadyPlayed/AlreadyPlayed'
```
Comment (DavideSilva) [inline frontend/src/App.jsx:9]: Is this meant to be here?

**Review by DavideSilva — APPROVED** [review]

**Pull request merged**